code/GridFieldOrderableRows.php

```diff
@@ -505,11 +508,15 @@ protected function reorderItems($list, array $values, array $sortedIDs) {
 			// similar to the SiteTree where you change the position, and then
 			// you go into the record and publish it.
 			foreach($sortedIDs as $sortValue => $id) {
-				if($map[$id] != $sortValue) {
-					$record = $class::get()->byID($id);
+				$record = $class::get()->byID($id);
```

Comment:
It seems like this could be left inside the condition below? It's a sliiiight performance impact moving it out of the condition because you're loading a DataObject for every item regardless of the following condition...